javascript/src/URDFClasses.js

```diff
@@ -237,9 +244,46 @@ class URDFJoint extends URDFBase {
             }
 
             case 'floating':
-            case 'planar':
                 // TODO: Support these joint types
                 console.warn(`'${ this.jointType }' joint not yet supported`);
+                break;
+            case 'planar':
+                // no-op if all values are identical to existing value
+                // TODO: this generic implementation of this check could be hoisted higher in the function if we want?
+                if (this.jointValue.every((value, index) => values[index] === value)) return didUpdate;
```

Comment:
I think it's okay to keep this here for now. We should also be clamping the values to the joint limit range and checking whether they changed after that. And if we're treating the arguments as "null === no change" then we want to make sure that's accounted for here, as well.

Reply:
The URDF spec is a little dubious on how limits work for multi-dof joints.

It says:
> \<limit> (required only for revolute and prismatic joint)

and doesn't seem to actually support limits in multiple axes.

Do we want to implement limits here, and have it be the same limit for both axes, or should we not implement a limit?

Reply:
Oh I missed that - yeah lets stick to the spec, then, and only use limits for revolute and prismatic joints.